ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala

```diff
@@ -235,75 +234,78 @@ private[indexer] class JdbcIndexer private[indexer] (
     new SubscriptionResourceOwner(readService)
 
   private def handleStateUpdate(
-      offset: Offset,
-      update: Update,
-  )(implicit loggingContext: LoggingContext): Future[Unit] = {
-    lastReceivedRecordTime = update.recordTime.toInstant.toEpochMilli
-
-    logger.trace(update.description)
-
-    metrics.daml.indexer.lastReceivedRecordTime.updateValue(lastReceivedRecordTime)
-    metrics.daml.indexer.lastReceivedOffset.updateValue(offset.toApiString)
-
-    val result = update match {
-      case PartyAddedToParticipant(
-          party,
-          displayName,
-          hostingParticipantId,
-          recordTime,
-          submissionId,
-          ) =>
-        val entry = PartyLedgerEntry.AllocationAccepted(
-          submissionId,
-          recordTime.toInstant,
-          domain.PartyDetails(party, Some(displayName), participantId == hostingParticipantId)
-        )
-        ledgerDao.storePartyEntry(offset, entry)
+      implicit loggingContext: LoggingContext): Flow[(Offset, Update), Unit, NotUsed] =
+    Flow[(Offset, Update)]
+      .wireTap(Sink.foreach[(Offset, Update)] {
+        case (offset, update) =>
+          lastReceivedRecordTime = update.recordTime.toInstant.toEpochMilli
+
+          logger.trace(update.description)
+
+          metrics.daml.indexer.lastReceivedRecordTime.updateValue(lastReceivedRecordTime)
+          metrics.daml.indexer.lastReceivedOffset.updateValue(offset.toApiString)
+      })
+      .mapAsync(1) {
```

Comment:
```suggestion
      .mapAsync(8) {
```

This block can safely run in parallel, no?

Reply:
I don't think it can. This would attempt at writing in parallel two transactions, with one with an older offset possibly being persisted before a newer one.

Reply:
If we split out writing the ledger end (as in #7845) and keep that sequential, it might be OK to do this in parallel.

Reply:
Hmm, would it @stefanobaghino-da ? This step prepares the inserts in a separate thread pool, with the results sent downstream in the same order they arrived. @rautenrieth-da , this is on my radar as well, but I would leave it for a different PR so I can track improvements in interative benchmarks so I can see how do they compose.

Reply:
@SamirTalwar-DA Can we ? You would have multiple transactions writing entries to the same tables at the same time

Reply:
> This would attempt at writing in parallel two transactions

As far as I can see, this block only prepares SQL statements, it does not do any database IO. And `mapAsync` should preserve the order of elements.

Reply:
Just for reference, @rautenrieth-da is referring to the `mapAsync` at line 248. Github screwed this up after an update of mine. To answer Robert's question, increasing parallelization here is on my radar but I would do it on a different PR in order to see how improvements compose. This PR already yielded a significant throughput improvement

Reply:
> > This would attempt at writing in parallel two transactions
> 
> As far as I can see, this block only prepares SQL statements, it does not do any database IO. And `mapAsync` should preserve the order of elements.

Uh, my bad. I even checked twice that I was commenting the appropriate line...

> Just for reference, @rautenrieth-da is referring to the `mapAsync` at line 248. Github screwed this up after an update of mine. To answer Robert's question, increasing parallelization here is on my radar but I would do it on a different PR in order to see how improvements compose. This PR already yielded a significant throughput improvement

Uh, I see, thanks.

---

Going back to the original topic: running multiple query preparations in parallel: since they have to be committed in order and right now the majority of the cost is persistence, I doubt there's anything to gain from doing that. Once preparation starts taking more time than persistence, there may be an advantage in doing that. If we want to experiment with this in the future I would recommend making this configurable and not put hard-coded values, forcing us to recompile everything to perform experiments.